routing/endpointregistry.go

```diff
@@ -115,16 +151,71 @@ func (r *EndpointRegistry) Do(routes []*Route) []*Route {
 	return routes
 }
 
+func (r *EndpointRegistry) updateStats() {
+	if !r.passiveHealthCheckEnabled {
+		return
+	}
+	ticker := time.NewTicker(r.statsResetPeriod)
+
+	for {
+		r.data.Range(func(key, value any) bool {
+			e := value.(*entry)
+
+			curSlot := e.curSlot.Load()
+			nextSlot := (curSlot + 1) % 2
+			e.totalFailedRoundTrips[nextSlot].Store(0)
+			e.totalRequests[nextSlot].Store(0)
+
+			failed := e.totalFailedRoundTrips[curSlot].Load()
+			requests := e.totalRequests[curSlot].Load()
+			failedRoundTripsRatio := float64(failed) / float64(requests)
+			// Given the current implementation, I expect failedRoundTripsRatio to be always less than 1
+			// because both failed and total requests counter may only increase over time and failed is loaded
+			// before total requests. However, after some code refactoring this assumption may not hold anymore,
+			// that's why I'm adding this line.
+			failedRoundTripsRatio = min(failedRoundTripsRatio, 1.0)
+			if requests > r.minRequests {
+				e.healthCheckDropProbability.Store(min(failedRoundTripsRatio, r.maxHealthCheckDropProbability))
+			} else {
+				e.healthCheckDropProbability.Store(0.0)
+			}
+			e.curSlot.Store(nextSlot)
+
+			return true
+		})
+
+		select {
+		case <-r.quit:
+			return
+		case <-ticker.C:
+		}
+	}
+}
+
+func (r *EndpointRegistry) Close() {
+	close(r.quit)
+}
+
 func NewEndpointRegistry(o RegistryOptions) *EndpointRegistry {
 	if o.LastSeenTimeout == 0 {
 		o.LastSeenTimeout = defaultLastSeenTimeout
 	}
 
-	return &EndpointRegistry{
-		data:            sync.Map{},
-		lastSeenTimeout: o.LastSeenTimeout,
-		now:             time.Now,
+	registry := &EndpointRegistry{
+		lastSeenTimeout:               o.LastSeenTimeout,
+		passiveHealthCheckEnabled:     o.PassiveHealthCheckEnabled,
+		statsResetPeriod:              o.StatsResetPeriod,
+		minRequests:                   o.MinRequests,
+		maxHealthCheckDropProbability: o.MaxHealthCheckDropProbability,
+
+		quit: make(chan struct{}),
+
+		now:  time.Now,
+		data: sync.Map{},
 	}
+	go registry.updateStats()
```

Comment:
I guess we want to check here the `o.PassiveHealthCheckEnabled` and remove the check and member `passiveHealthCheckEnabled` in the registry.